src/data/bucket/fill_extrusion_bucket.js

```diff
@@ -219,11 +221,6 @@ class FillExtrusionBucket implements Bucket {
                 segment = this.segments.prepareSegment(numVertices, this.layoutVertexArray, this.indexArray);
             }
 
-            //Only triangulate and draw the area of the feature if it is a polygon
-            //Other feature types (e.g. LineString) do not have area, so triangulation is pointless / undefined
-            if (vectorTileFeatureTypes[feature.type] !== 'Polygon')
-                continue;
```

Comment:
The position of this check prevents constructing geometry for caps (roofs, ceilings), and also prevents attempting to triangulate this cap geometry. Do we remove this check because the two following for loops and earcut() effectively `noop` in this case?

Reply:
I ran the render tests with this check re-included and it produces the old test-case output that has been updated elsewhere in this PR. So this check does change the output. No idea if we like that output but we did make a test explicitly for this case of `MultiLineString` feature type in a fill-extrusion bucket.

Reply:
I don't have the context for this either, but `openmaptiles` does have some of their buildings as `LineString` and this check was preventing only the caps from rendering.

I'm thinking we can either not generate the entire `fill-extrusion` when the feature type isn't `Polygon` or, we can do what this PR does and tries to do and ignores the check, or keep the status quo.

Reply:
I see, so we have closed `Polygon` data mislabeled as `LineString`. I'm assuming this happens in the GeoJSON or tiles themselves. I think if you saw more output that looks like correct buildings (newly generated with caps) by removing this check, then maybe removing the check is the right thing to do. I think I agree that we should either render the cap or render no extrusion at all. We should get some more feedback from people who know the user input data better.

Reply:
This check was added in https://github.com/mapbox/mapbox-gl-js/pull/7685 to avoid visual artifacts. @mourner or @ansis should have more context

Reply:
Thanks for the issue link @ryanhamley, I reverted the change. It seems like it seems like its a good idea to leave it in.

Reply:
I still want to know if these caps we're avoiding would be wrong artifacts or potentially representing data meant as a polygon that wasn't correctly labeled, but agree that reverting is probably best.